File: spendzero/mobile/lib/design_system/gradients.dart
import 'package:flutter/widgets.dart';

import 'colors.dart';

/// Gradients reproduced exactly from inline `style={{ background: ... }}`
/// in the Lovable source.
class DSGradients {
  DSGradients._();

  /// Home ambient wash:
  /// radial(120% 60% at 50% 0%, gold/10, transparent 60%),
  /// radial(80% 40% at 20% 100%, future/8, transparent 70%).
  static const Color ambientGold = Color(0x1AD8B36A); // gold @ 10%
  static const Color ambientFuture = Color(0x144DA3FF); // future @ 8%

  /// Hero image scrim:
  /// linear 180deg, bg 15% -> 25% (35%) -> 85% (78%) -> 98% (100%).
  static const List<Color> heroScrim = [
    Color(0x260A0B0E), // 15%
    Color(0x400A0B0E), // 25%
    Color(0xD90A0B0E), // 85%
    Color(0xFA0A0B0E), // 98%
  ];
  static const List<double> heroScrimStops = [0.0, 0.35, 0.78, 1.0];

  /// Dream card scrim: linear 180deg, 5% -> 55% (60%) -> 95% (100%).
  static const List<Color> dreamCardScrim = [
    Color(0x0D0A0B0E),
    Color(0x8C0A0B0E),
    Color(0xF20A0B0E),
  ];
  static const List<double> dreamCardScrimStops = [0.0, 0.6, 1.0];

  /// Gold progress fill: linear 90deg, gold -> #f3dfa6.
  static const LinearGradient goldFill = LinearGradient(
    begin: Alignment.centerLeft,
    end: Alignment.centerRight,
    colors: [DSColors.gold, Color(0xFFF3DFA6)],
  );

  /// Bottom nav center-action radial button.
  static const RadialGradient navCenterAction = RadialGradient(
    center: Alignment(-0.4, -0.4),
    colors: [Color(0xFFF5E1AA), DSColors.gold, Color(0xFFA8853D)],
    stops: [0.0, 0.55, 1.0],
  );

  /// Hero card "Active Dream" open-button radial accent (NavBar variant).
  static const RadialGradient heroOpenAccent = RadialGradient(
    center: Alignment(-0.4, -0.4),
    colors: [Color(0xFFEBC98C), Color(0xFFB8923F)],
  );

  /// Bottom nav glass background: linear 180deg surfaceElevated/85 -> surface/85.
  static const LinearGradient navGlass = LinearGradient(
    begin: Alignment.topCenter,
    end: Alignment.bottomCenter,
    colors: [Color(0xD91A1D24), Color(0xD9111318)],
  );

  /// Weekly momentum bar: non-peak day fill.
  static const LinearGradient momentumBarIdle = LinearGradient(
    begin: Alignment.topCenter,
    end: Alignment.bottomCenter,
    colors: [Color(0x40FFFFFF), Color(0x14FFFFFF)],
  );

  /// Weekly momentum bar: peak day fill.
  static const LinearGradient momentumBarPeak = LinearGradient(
    begin: Alignment.topCenter,
    end: Alignment.bottomCenter,
    colors: [DSColors.gold, Color(0xFFB8923F)],
  );

  /// `text-shimmer-gold`: gold-soft -> gold -> #fff3d6 -> gold -> gold-soft.
  static const List<Color> textShimmerGold = [
    DSColors.goldSoft,
    DSColors.gold,
    Color(0xFFFFF3D6),
    DSColors.gold,
    DSColors.goldSoft,
  ];
  static const List<double> textShimmerStops = [0.0, 0.4, 0.5, 0.6, 1.0];
}

File: spendzero/mobile-rn/design-system/gradients.ts
import { colors } from "./colors";

/** Gradients reproduced exactly from inline `style={{ background: ... }}` in the Lovable source. */
export const gradients = {
  /** Home ambient wash: radial(gold/10) top, radial(future/8) bottom-left. */
  ambientGold: "rgba(216,179,106,0.10)",
  ambientFuture: "rgba(77,163,255,0.08)",

  /** Hero image scrim: linear 180deg, bg 15% -> 25% (35%) -> 85% (78%) -> 98% (100%). */
  heroScrim: {
    colors: [
      "rgba(10,11,14,0.15)",
      "rgba(10,11,14,0.25)",
      "rgba(10,11,14,0.85)",
      "rgba(10,11,14,0.98)",
    ] as const,
    locations: [0, 0.35, 0.78, 1] as const,
  },

  /** Dream card scrim: linear 180deg, 5% -> 55% (60%) -> 95% (100%). */
  dreamCardScrim: {
    colors: ["rgba(10,11,14,0.05)", "rgba(10,11,14,0.55)", "rgba(10,11,14,0.95)"] as const,
    locations: [0, 0.6, 1] as const,
  },

  /** Gold progress fill: linear 90deg, gold -> #f3dfa6. */
  goldFill: { colors: [colors.gold, "#F3DFA6"] as const },

  /** Bottom nav center-action radial button. */
  navCenterAction: {
    colors: ["#F5E1AA", colors.gold, "#A8853D"] as const,
    locations: [0, 0.55, 1] as const,
  },

  /** Bottom nav glass background: linear 180deg surfaceElevated/85 -> surface/85. */
  navGlass: {
    colors: ["rgba(26,29,36,0.85)", "rgba(17,19,24,0.85)"] as const,
  },

  /** Weekly momentum bar: non-peak day fill. */
  momentumBarIdle: {
    colors: ["rgba(255,255,255,0.25)", "rgba(255,255,255,0.08)"] as const,
  },

  /** Weekly momentum bar: peak day fill. */
  momentumBarPeak: {
    colors: [colors.gold, "#B8923F"] as const,
  },

  /** `text-shimmer-gold`: gold-soft -> gold -> #fff3d6 -> gold -> gold-soft. */
  textShimmerGold: {
    colors: [colors.goldSoft, colors.gold, "#FFF3D6", colors.gold, colors.goldSoft] as const,
    locations: [0, 0.4, 0.5, 0.6, 1] as const,
  },
} as const;
